ModelFitting: Fixes related to NaN values on the weight

ayllon opened the pull request:
Fixes #326.

Two changes here, really

1. Write into the catalog the stop reason, and flag as ERROR if the stop reason is not due to convergence (i.e. 7 = NaN or Inf)
2. Clip negative flux values to 0, avoiding NaN on the weight map

marcschefer commented on SEImplementation/src/lib/Plugin/FlexibleModelFitting/FlexibleModelFittingTask.cpp:
```diff
@@ -104,6 +104,14 @@ void printLevmarInfo(std::array<double, 10> info) {
 }
 
+static void updateFlags(Flags& flags, unsigned stop_reason) {
+  switch (stop_reason) {
```
I have noticed the code for the GSL integration will return 0 as stop reason in case of unexpected result as well as a few levmar-compatible codes, maybe it should also be added to the possible errors. Or maybe this should reverse logic, expected stop reasons are ok everything else is considered an error?

ayllon replied:
True. Another option would be to have a Flag on the `LeastSquareSummary` struct and let the engine-specific implementation decide?

marcschefer replied:
Yes, I think it's best to be flexible for future engine implementations

ayllon commented:
I have added an extra field to hold the engine-dependent "stop reason" so there is no need to cast the `underlying_framework_info` anymore. This should be more portable.

I have also replaced the "success flag", which, by the way, was not read, with an enum that allows to differentiate between an error, a success, and max.iterations reached. And now it is used to set up the `fmf_flag`.

I wasn't sure if we want to flag the max.iterations on the output catalog. I have left it as it was, but at least know we have the possibility.
